sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_appinsights_utils.py

```diff
@@ -31,6 +31,28 @@ def default_resource_group_for_app_insights_exists(
         return False
 
 
+def clean_up_rg_deployments_on_subscription(
+    credentials: TokenCredential, subscription_id: str
+):
+    client = ResourceManagementClient(
+        credential=credentials,
+        subscription_id=subscription_id,
+        base_url=_get_base_url_from_metadata(),
+        api_version=ArmConstants.AZURE_MGMT_RESOURCE_API_VERSION,
+    )
+    # get all deployments that are succeeded with DeployResourceGroup pattern
+    deployments = client.deployments.list_at_subscription_scope(filter="provisioningState eq 'Succeeded'")
+    rg_deployments = [d.name for d in list(deployments) if "DeployResourceGroup" in d.name]
```

Comment:
Is there a way to tell if these RG deployments were initiated by AzureML or someone else? We shouldn't touch deployments' initiated by others.

Reply:
I am only cleaning up RG deployments that have the DeployResourceGroup in the name, which is the pattern used for deployments from SDKv2 template. I updated to AzureMLDefaultResourceGroupDeployment to make it even more clear.